### PR TITLE
Create Eveonline 0uath2 provider

### PR DESCRIPTION
**Pull request by Grasume**
This Pr is for the addition of the Eveonline provider.

### CODE REVIEW

**Comment by Grasume** [comment]
@atymic are you able to review and approve ?

File: src/Eveonline/composer.json [hunk 1]
```diff
@@ -0,0 +1,21 @@
+{
+    "name": "socialiteproviders/eveonline",
+    "description": "Eveonline OAuth2 Provider for Laravel Socialite",
+    "license": "MIT",
+    "authors": [{
+        "name": "James Gerstenberg",
+        "email": "grasume0@gmail.com"
+    }],
+    "license": "MIT",
```
Comment (lucasmichot) [inline src/Eveonline/composer.json:9]: The license entry is duplicated.

Comment (Grasume) [reply src/Eveonline/composer.json]: @lucasmichot Good catch fixed it.

File: src/Eveonline/Provider.php [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+    protected $scopeSeparator = ' ';
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getAuthUrl($state)
+    {
+        $endpoint = 'https://login.eveonline.com/v2/oauth/authorize/';
+
+        return$this->buildAuthUrlFromBase($endpoint, $state);
```
Comment (lucasmichot) [inline src/Eveonline/Provider.php:41]:
```suggestion
        return $this->buildAuthUrlFromBase('https://login.eveonline.com/v2/oauth/authorize/', $state);
```

File: src/Eveonline/Provider.php [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getTokenUrl()
+    {
+        $endpoint = 'https://login.eveonline.com/v2/oauth/token';
+
+        return $endpoint;
```
Comment (lucasmichot) [inline src/Eveonline/Provider.php:51]:
```suggestion
        return 'https://login.eveonline.com/v2/oauth/token';
```

**Review by atymic — APPROVED** [review]

**Pull request merged**